source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -108,7 +108,7 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         auto elementType = inst->getElementType();
         IRSizeAndAlignment elementSize;
         getSizeAndAlignment(m_sharedContext->m_targetProgram->getOptionSet(), layoutRules, elementType, &elementSize);
-        elementSize = layoutRules->alignCompositeElement(elementSize);
+        elementSize = layoutRules->alignCompositeElementOfNonAggregate(elementSize);
```

Comment:
why is this nonAggregate?

Reply:
If this is `nonAggregate`, then `_calcArraySizeAndAlignment` should also call `alignCompositeElementOfNonAggregate`?

Reply:
Should be `alignCompositeElementOfAggregate` 👍